app/src/main/java/com/pajato/android/gamechat/database/ExperienceManager.java

```diff
@@ -289,6 +290,13 @@ public void updateExperience(final Experience experience) {
         String groupKey = experience.getGroupKey();
         String roomKey = experience.getRoomKey();
         String expKey = experience.getExperienceKey();
+        // Handle offline game play
```

Comment:
I have found that snippets like this can be more readable by doing something like:

    // Persist the experience and handle off-line behavior by short-circuiting the Firebase update.
    ....
    String expKey = ...;
    if (groupKey == null &&  roomKey == null && expKey.equals(OFFLINE_EXPERIENCE_KEY)) {
        AppEventManager.instance.pos(new ExperienceChangeEvent(experience, CHANGED);
        return;
    }

    // Handle the normal (net connected) update.
    ...

The point is to  collect comments and code in short concise snippets and to express statements in a single line while still making analysis and debugging easy.  Fwiw.